shadows/framework/src/main/java/org/robolectric/android/controller/ComponentController.java

```diff
@@ -20,6 +21,13 @@
   @SuppressWarnings("unchecked")
   public ComponentController(T component, Intent intent) {
     this(component);
+    if (intent != null) {
```

Comment:
Can you extract this into a function and also call it from `ActivityController.onNewIntent`?

Reply:
Also, do you think it makes sense to only do this when there are extras, which are susceptible to Parcel/Unparcel issues? I do not think there are other user-defined data in an intent outside of extras that could cause parcel/unparcel issues. So this would be:

```java
if (intent != null || intent.getExtras() != null || intent.getExtras().size() > 0) {
      ...
    }
```

Reply:
> Also, do you think it makes sense to only do this when there are extras, which are susceptible to Parcel/Unparcel issues? I do not think there are other user-defined data in an intent outside of extras that could cause parcel/unparcel issues. So this would be:
> 
> ```java
> if (intent != null || intent.getExtras() != null || intent.getExtras().size() > 0) {
>       ...
>     }
> ```

I think we should try to emulate the Android system as much as possible, so Parcel/Unparcel it in all cases.